src/krbd.cc

```diff
@@ -241,60 +243,78 @@ static int wait_for_udev_add(struct udev_monitor *mon, const krbd_spec& spec,
   for (;;) {
     struct pollfd fds[1];
     struct udev_device *dev;
-    int r;
 
     fds[0].fd = udev_monitor_get_fd(mon);
     fds[0].events = POLLIN;
     r = poll(fds, 1, POLL_TIMEOUT);
-    if (r < 0)
-      return -errno;
+    if (r <= 0) {
+      r = (r == 0) ? -ETIMEDOUT : -errno;
+      break;
+    }
 
-    if (r == 0)
-      return -ETIMEDOUT;
 
     dev = udev_monitor_receive_device(mon);
     if (!dev)
       continue;
 
-    if (strcmp(udev_device_get_action(dev), "add") != 0)
-      goto next;
+    if (strcmp(udev_device_get_action(dev), "add") != 0) {
+      udev_device_unref(dev);
+      continue;
+    }
 
-    if (!bus_dev) {
-      if (strcmp(udev_device_get_subsystem(dev), "rbd") == 0) {
+    const char* sub_sys = udev_device_get_subsystem(dev);
```

Comment:
No need for this variable, let `udev_device_get_subsystem` be called twice, as before.